api/test/e2e/consumer_test.go

```diff
@@ -0,0 +1,214 @@
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "\"username\":\"jack\",\"desc\":\"test description\",\"plugins\":{\"key-auth\":{\"key\":\"auth-two\"}},\"labels\":{\"build\":\"16\",\"env\":\"production\",\"version\":\"v2\"}",
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc: "create the route",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"plugins": {
+					"key-auth": {}
+				},
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": [{
+						"host": "172.16.238.20",
+						"port": 1980,
+						"weight": 1
+					}]
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "hit the route with correct apikey",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"apikey": "auth-two"},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc:     "delete the consumer",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/consumers/jack",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "delete the route",
+			Object:       MangerApiExpect(t),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+	}
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+}
+
+func TestConsumer_with_createtime_updatetime(t *testing.T) {
+	//create consumer
+	basepath := "http://127.0.0.1:8080/apisix/admin/consumers"
+	data := `{
+		"username":"jack",
+		"desc": "new consumer"
+    }`
+	request, _ := http.NewRequest("PUT", basepath, strings.NewReader(data))
+	request.Header.Add("Authorization", token)
+	resp, _ := http.DefaultClient.Do(request)
+	defer resp.Body.Close()
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	code := gjson.Get(string(respBody), "code")
+	assert.Equal(t, code.String(), "0")
+
+	time.Sleep(time.Duration(100) * time.Millisecond)
+	
+	//get the consumer, save createtime and updatetime
+	request, _ = http.NewRequest("GET", basepath+"/jack", nil)
+	request.Header.Add("Authorization", token)
+	resp, _ = http.DefaultClient.Do(request)
+	respBody, _ = ioutil.ReadAll(resp.Body)
+	createtime := gjson.Get(string(respBody), "data.create_time")
+	updatetime := gjson.Get(string(respBody), "data.update_time")
+
+	//Wait one second for update time to be different
+	time.Sleep(time.Duration(1000) * time.Millisecond)
+
+	//update the consumer with new desc
+	data = `{
+		"username":"jack",
+		"desc": "updated consumer"
+    }`
+	request, _ = http.NewRequest("PUT", basepath, strings.NewReader(data))
+	request.Header.Add("Authorization", token)
+	resp, _ = http.DefaultClient.Do(request)
```

Comment:
After we deleted them, we need to make a confirm at the data plane.

Reply:
fixed